test/pp_api_test/test_feature_extraction.cc

```diff
@@ -99,3 +99,78 @@ TEST(ExtractorTest, TestPhi4AudioFeatureExtraction8k) {
 }
+
+TEST(ExtractorTest, TestPhi4AudioOutput) {
+  const char* audio_path[] = {"data/1272-141231-0002.wav"};
+  OrtxObjectPtr<OrtxRawAudios> raw_audios;
+  extError_t err = OrtxLoadAudios(raw_audios.ToBeAssigned(), audio_path, 1);
+  ASSERT_EQ(err, kOrtxOK);
+
+  OrtxObjectPtr<OrtxFeatureExtractor>
+    feature_extractor(OrtxCreateSpeechFeatureExtractor, "data/models/phi-4/audio_feature_extraction.json");
+  OrtxObjectPtr<OrtxTensorResult> result;
+  err = OrtxFeatureExtraction(feature_extractor.get(), raw_audios.get(), result.ToBeAssigned());
+  ASSERT_EQ(err, kOrtxOK);
+
+  OrtxObjectPtr<OrtxTensor> tensor;
+  err = OrtxTensorResultGetAt(result.get(), 0, tensor.ToBeAssigned());
+  ASSERT_EQ(err, kOrtxOK);
+
+  const float* data{};
+  const int64_t* shape{};
+  size_t num_dims;
+  err = OrtxGetTensorData(tensor.get(), reinterpret_cast<const void**>(&data), &shape, &num_dims);
+  ASSERT_EQ(err, kOrtxOK);
+  ASSERT_EQ(std::vector<int64_t>(shape, shape + num_dims), std::vector<int64_t>({1, 1332, 80}));
+
+  // Dimensions
+  const size_t num_rows = shape[1];
+  const size_t num_columns = shape[2];
+
+  // Read the expected output from the file
+  std::filesystem::path expected_file_path = "data/models/phi-4/expected_output.txt";
+  std::ifstream infile(expected_file_path);
+
+  if (!infile.is_open()) {
+      std::cerr << "Failed to open expected output file: " << expected_file_path << std::endl;
+      return;
+  }
+
+  // Define lambda for comparison
+  auto are_close = [](float a, float b, float rtol = 1e-03, float atol = 1e-02) -> bool {
+      return std::abs(a - b) <= atol || std::abs(a - b) <= rtol * std::abs(b);
+  };
+
+  size_t num_mismatched = 0;
+  size_t total_elements = num_rows * 10;  // We only compare the first 10 columns
+  std::string line;
+  size_t row_idx = 0;
+
+  while (std::getline(infile, line) && row_idx < num_rows) {
+      std::stringstream ss(line); // Stringstream to parse each line
+      std::string value_str;
+      size_t col_idx = 0;
+
+      while (std::getline(ss, value_str, ',') && col_idx < 10) {  // Only read the first 10 columns
+          float expected_value = std::stof(value_str);  // Convert string to float
+
+          // Compare values
+          const float* row_start = data + (row_idx * num_columns);
+          if (!are_close(row_start[col_idx], expected_value)) {
+              num_mismatched++;  // Count mismatches
+              std::cout << "Mismatch at (" << row_idx << "," << col_idx << "): "
+                        << "Expected: " << expected_value << ", Got: " << row_start[col_idx] << std::endl;
+          }
+          col_idx++;
+      }
+      row_idx++;
+  }
+
+  infile.close();
+
+  // Calculate the mismatch percentage
+  float mismatch_percentage = static_cast<float>(num_mismatched) / total_elements;
+
+  std::cout << "Mismatch percentage: " << mismatch_percentage * 100 << "%" << std::endl;
+  ASSERT_LT(mismatch_percentage, 0.02) << "Mismatch percentage exceeds 2% threshold!";
```

Comment:
Where does this percentage come from?

Reply:
Same as whisper python unit test here: https://github.com/microsoft/onnxruntime-extensions/blob/main/test/test_whisper.py#L61

HF has 1.5% mismatch; our mismatch percentage in this example is around 0.07%.